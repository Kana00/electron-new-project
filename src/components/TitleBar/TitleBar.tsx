import * as React from 'react';
import * as os from 'os';
import Flexbox from 'flexbox-react';
import configApp from '../../appconfigrc';
import './TitleBar';
import WindowsWindowControl from './WindowsWindowControl/WindowsWindowControl';


export default class TitleBar extends React.Component {
  private _operatingSystem: string;
  private _heightOfTitleBar: string;
  private _paddingInterElement: string;
  constructor(props: any) {
    super(props);
    this._operatingSystem = os.type();
    this._operatingSystem = "Windows_NT";
  }

  render() {
    if(this._operatingSystem === "Darwin") {
      const styles = {
        titleContainer: {
          backgroundColor: '#282C34',
          height: '22px',
          width: '100vw',
          WebkitUserSelect: 'none',
          WebkitAppRegion: 'drag',
          Display: 'flex',
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
        },
        titleOfWindow: {
          fontFamily: 'Helvetica',
          fontSize: 14,
          color: '#9DA5B4',
        }
      }
      return (
        <div style={styles.titleContainer}>
          <span style={styles.titleOfWindow}>{configApp.window.title}</span>
        </div>
      );
    } else if (this._operatingSystem === "Linux" || this._operatingSystem === "Windows_NT") {
      this._heightOfTitleBar = '30px';
      this._paddingInterElement = '10px';
      const styles = {
        titleContainer: {
          backgroundColor: '#282C34',
          height: this._heightOfTitleBar,
          WebkitUserSelect: 'none',
          WebkitAppRegion: 'drag',
          fontWeight: '100',
        },
        controlRight: {

        },
        controlLeft: {
          fontFamily: 'Helvetica',
          fontSize: 12,
          color: '#9DA5B4',
          lineHeight: this._heightOfTitleBar,
          textAlign: 'center',
        },
        elementOnBar: {
          paddingLeft: this._paddingInterElement,

        }
      }
      return (
        <Flexbox flexDirection="row" justifyContent="space-between" style={styles.titleContainer}>
          <div style={styles.controlLeft}>
            <div style={styles.elementOnBar}>
              <span>{configApp.window.title}</span>
            </div>
          </div>
          <div style={styles.controlRight}>
            <WindowsWindowControl colorOfControl="#BBBBBB" onOverColor="#EEEEEE" height={this._heightOfTitleBar}/>
          </div>
        </Flexbox>
      )
    } else {
      return (<div><p>No detected operating system (NavigationBar react component)</p></div>)
    }
  }
}
