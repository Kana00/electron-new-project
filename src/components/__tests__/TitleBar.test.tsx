import * as React from 'react';
import TitleBar from '../TitleBar/TitleBar';
import * as renderer from 'react-test-renderer';
import * as Adapter from 'enzyme-adapter-react-16';
import { configure as configureEnzyme } from 'enzyme';
import { shallow } from 'enzyme';

configureEnzyme({ adapter: new Adapter() });

describe('Test component TitleBar', () => {
  it('render correctly', () => {
    const tree = renderer.create(<TitleBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('contains WindowsWindowControl', () => {
    const wrapper = shallow(<TitleBar />);
    expect(wrapper.find('WindowsWindowControl').length).toBe(1);
  });
});
