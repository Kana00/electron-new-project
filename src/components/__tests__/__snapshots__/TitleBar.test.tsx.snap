// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component TitleBar render correctly 1`] = `
<div
  className="sc-bdVaJa cmuWdh"
  style={
    Object {
      "WebkitAppRegion": "drag",
      "WebkitUserSelect": "none",
      "backgroundColor": "#282C34",
      "fontWeight": "100",
      "height": "30px",
    }
  }
>
  <div
    style={
      Object {
        "color": "#9DA5B4",
        "fontFamily": "Helvetica Neue",
        "fontSize": 14,
        "lineHeight": "30px",
        "textAlign": "center",
      }
    }
  >
    <div
      style={
        Object {
          "paddingLeft": "10px",
        }
      }
    >
      <span>
        My new project
         âŽª File___Edit___View___Help
      </span>
    </div>
  </div>
  <div
    style={Object {}}
  >
    <div
      className="sc-bdVaJa dkRebT"
      style={
        Object {
          "color": "#BBBBBB",
          "fontSize": "15px",
          "lineHeight": "30px",
        }
      }
    >
      <div
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "paddingLeft": "10px",
            "paddingRight": "10px",
          }
        }
      >
        <span
          className="ti-minus"
        />
      </div>
      <div
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "paddingLeft": "10px",
            "paddingRight": "10px",
          }
        }
      >
        <span
          className="ti-plus"
        />
      </div>
      <div
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "paddingLeft": "10px",
            "paddingRight": "10px",
          }
        }
      >
        <span
          className="ti-close"
        />
      </div>
    </div>
  </div>
</div>
`;
